packages/ui/src/ui/store/selectors/settings/settings-queries.ts

```diff
@@ -1,6 +1,39 @@
 import {createSelector} from 'reselect';
 import {getSettingsData} from './settings-base';
+import {getCluster} from '../global';
 
 export const getQuerySuggestionsEnabled = createSelector(getSettingsData, (data) => {
     return data['global::queryTracker::suggestions'] || false;
 });
+
+export const getFavoriteQueryEngine = createSelector(
+    [getSettingsData, getCluster],
+    (data, cluster) => {
+        return data[`local::${cluster}::queryTracker::favoriteEngine`];
+    },
+);
+
+export const getFavoriteQueryPath = createSelector(
+    [getSettingsData, getCluster],
+    (data, cluster) => {
+        return data[`local::${cluster}::queryTracker::favoritePath`];
+    },
+);
+
+export const getFavoriteQueryClique = createSelector(
+    [getSettingsData, getCluster],
+    (data, cluster) => {
+        return data[`local::${cluster}::queryTracker::favoriteClique`];
+    },
+);
+
+export const getFavoriteQuerySettings = createSelector(
```

Comment:
It is more clear when selectors are used separately. There is no reason to have such combining selectors without any transformation of incoming parameters.